server/metadata/encoding/dictionary.go

```diff
@@ -286,18 +301,123 @@ func (k *DictionaryEncoder) validCollectionId(id uint32) error {
 	return nil
 }
 
-func (k *DictionaryEncoder) getDatabaseId(ctx context.Context, tx transaction.Tx, dbName string, namespaceId uint32) (uint32, error) {
-	key := keys.NewKey(encodingSubspaceKey, encVersion, UInt32ToByte(namespaceId), dbKey, dbName, keyEnd)
+func (k *DictionaryEncoder) GetDatabases(ctx context.Context, tx transaction.Tx, namespaceId uint32) (map[string]uint32, error) {
+	databases := make(map[string]uint32)
+	it, err := tx.Read(ctx, keys.NewKey(EncodingSubspaceKey, encVersion, UInt32ToByte(namespaceId), dbKey))
+	if err != nil {
+		return nil, err
+	}
+
+	var v kv.KeyValue
+	for it.Next(&v) {
+		if it.Err() != nil {
```

Comment:
This should be checked after the loop.
If `Next` returns `true` it's guaranteed that there is no error.

Reply:
This also means that error case is not tested :)

Reply:
moved it :)